### PR TITLE
ApiTableCatalogItem and points time-series fixes

### PR DESCRIPTION
**Pull request by nf-s**
### ApiTableCatalogItem and points time-series fixes

Related to https://github.com/TerriaJS/terriajs/issues/5470

* Move feature info timeseries chart funtion to `lib\Table\getChartDetailsFn.ts`
* Fix feature info timeseries chart for point (lat/long) timeseries
* Feature info chart x-values are now be sorted in acending order
* Remove merging rows by ID for `PER_ROW` data in `ApiTableCatalogItem`
* Make `ApiTableCatalogItem` more compatible with Table `Traits`
  * `keyToColumnMapping` has been removed, now columns must be defined in `columns` `TableColumnTraits` to be copied from API responses.

### Test URL

https://gist.githubusercontent.com/nf-s/845b4ec1c00d0142282394eba4b1a26c/raw/a49f9df540ffff06bc546fd7f7b328d2c633c8b2/api-tab-item-v2.json

### New JSON

```json
{
      "id": "6e650884-dea9-4c0e-8823-94e360238b66",
      "type": "api-table",
      "apis": [
        {
          "url": "https://data.melbourne.vic.gov.au/resource/d6mv-s43h.json?$limit=50000"
        },
        {
          "url": "https://data.melbourne.vic.gov.au/resource/h57g-5234.json",
          "kind": "PER_ID"
        }
      ],
      "description": "Pedestrian Traffic",
      "idKey": "sensor_id",
      "name": "Pedestrian Traffic",
      "refreshInterval": 60,
      "removeDuplicateRows": true,
      "defaultStyle": {
        "time": {
          "spreadStartTime": true,
          "timeColumn": "date_time"
        }
      },
      "activeStyle": "total_of_directions",
      "columns": [
        {
          "title": "Total in both directions",
          "name": "total_of_directions"
        },
        {
          "title": "Time",
          "name": "date_time"
        },
        {
          "title": "latitude",
          "name": "latitude"
        },
        {
          "title": "longitude",
          "name": "longitude"
        },
        {
          "title": "Sensor ID",
          "name": "sensor_id"
        },
        {
          "title": "Installation Date",
          "name": "installation_date"
        },
        {
          "name": "sensor_description",
          "title": "Sensor Description",
          "type": "hidden"
        }
      ]
    }
```

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: lib/ModelMixins/TableMixin.ts [hunk 1]
```diff
@@ -804,27 +805,12 @@ function TableMixin<T extends Constructor<Model<TableTraits>>>(Base: T) {
...
-            yName: this.activeTableStyle.colorColumn?.title,
-            units: chartColumns.map(column => column.units || ""),
-            csvData: [
-              chartColumns.map(col => col!.title).join(","),
-              ...regionIds.map(i =>
-                chartColumns.map(col => col.valueFunctionForType(i)).join(",")
-              )
-            ].join("\n")
-          });
+          featureInfo.properties._terria_getChartDetails = getChartDetailsFn(
```
Comment (nf-s) [inline lib/ModelMixins/TableMixin.ts:810]: This has been moved into `getChartDetailsFn`

File: lib/Table/TableStyle.ts [hunk 2]
```diff
@@ -499,43 +499,61 @@ export default class TableStyle {
...
+      const startDate = this.startJulianDates?.[i] ?? date;
+      const finishDate = this.finishJulianDates?.[i] ?? undefined;
 
-        const finishDate = getFinishDate(startDate, i);
-        return new TimeInterval({
-          start: startDate,
-          stop: finishDate,
-          isStopIncluded: JulianDate.equals(finishDate, lastDate),
-          data: startDate
-        });
```
Comment (nf-s) [inline lib/Table/TableStyle.ts:533]: Some of this has been moved into `finishJulianDates`

File: lib/Table/createLongitudeLatitudeFeaturePerId.ts [hunk 3]
```diff
@@ -82,20 +74,6 @@ function createFeature(
...
-  );
-  const getChartDetails = () => ({
-    title: featureId,
-    csvData: [
-      chartColumns.map(col => col.name).join(","),
-      ...rowIds.map(i =>
-        chartColumns.map(col => col.valueFunctionForType(i)).join(",")
-      )
-    ].join("\n")
-  });
```
Comment (nf-s) [inline lib/Table/createLongitudeLatitudeFeaturePerId.ts:98]: This has been moved into `getChartDetailsFn`

File: lib/Table/createLongitudeLatitudeFeaturePerId.ts [hunk 4]
```diff
@@ -33,15 +33,7 @@ type RequiredTableStyle = TableStyle & {
...
-  // Group rows by id and create a feature per group
-  const idColumns = style.idColumns;
-  const rowIds = style.tableModel.rowIds;
-  const groups = Object.entries(
-    groupBy(rowIds, id =>
-      idColumns.map(col => col.valueFunctionForType(id)).join("-")
-    )
-  );
-  const features = groups.map(([featureId, rowIds]) =>
+  const features = style.rowGroups.map(([featureId, rowIds]) =>
```
Comment (nf-s) [inline lib/Table/createLongitudeLatitudeFeaturePerId.ts:36]: This has been moved into `style.rowGroups`

**Review by KeyboardSounds — APPROVED** [review]

**Pull request merged**